From 179c845d7b5e0fa540da4204233e932e8f8d9423 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 27 Jul 2018 20:54:53 +0100
Subject: [PATCH 1/1] dbus: Teach it a new trick: Shutting up

There are times when the d-bus spam is monstrous, so let's teach it
to respect a new variable, DBUS_SILENCE_WARNINGS. If set, we'll not
hear any more about it.

This still ensures the internal safety checking takes place, it just
saves the user from a wall of spam when set.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 dbus/dbus-internals.c | 27 ++++++++++++++++++++-------
 1 file changed, 20 insertions(+), 7 deletions(-)

diff --git a/dbus/dbus-internals.c b/dbus/dbus-internals.c
index 267aef97..36660c7c 100644
--- a/dbus/dbus-internals.c
+++ b/dbus/dbus-internals.c
@@ -185,7 +185,8 @@ const char *_dbus_no_memory_message = "Not enough memory";

 static dbus_bool_t warn_initted = FALSE;
 static dbus_bool_t fatal_warnings = FALSE;
-static dbus_bool_t fatal_warnings_on_check_failed = TRUE;
+static dbus_bool_t fatal_warnings_on_check_failed = FALSE;
+static dbus_bool_t silence_warnings = FALSE;

 static void
 init_warnings(void)
@@ -213,6 +214,12 @@ init_warnings(void)
             }
         }
 
+      s = _dbus_getenv ("DBUS_SILENCE_WARNINGS");
+      if (s)
+        {
+          silence_warnings = TRUE;
+        }
+
       warn_initted = TRUE;
     }
 }
@@ -239,9 +246,12 @@ _dbus_warn (const char *format,
   if (fatal_warnings)
     severity = DBUS_SYSTEM_LOG_ERROR;
 
-  va_start (args, format);
-  _dbus_logv (severity, format, args);
-  va_end (args);
+  if (!silence_warnings)
+    {
+      va_start (args, format);
+      _dbus_logv (severity, format, args);
+      va_end (args);
+    }
 
   if (fatal_warnings)
     {
@@ -271,9 +281,12 @@ _dbus_warn_check_failed(const char *format,
   if (fatal_warnings_on_check_failed)
     severity = DBUS_SYSTEM_LOG_ERROR;
 
-  va_start (args, format);
-  _dbus_logv (severity, format, args);
-  va_end (args);
+  if (!silence_warnings)
+    {
+      va_start (args, format);
+      _dbus_logv (severity, format, args);
+      va_end (args);
+    }
 
   if (fatal_warnings_on_check_failed)
     {
-- 
2.18.0

